privacyIDEA Authenticator/TwoStepRollout.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  TwoStepRollout.swift
+//  privacyIDEA Authenticator
+//
+//  Created by Nils Behlen on 15.05.19.
+//  Copyright © 2019 Nils Behlen. All rights reserved.
+//
+
+import Foundation
+import CommonCrypto
+import SwiftOTP
+
+class TwoStepRollout {
+    
+    var listViewDelegate: TokenlistDelegate
+    
+    init(_ delegate: TokenlistDelegate) {
+        self.listViewDelegate = delegate
+    }
+    
+    // MARK: - 2STEP
+    func do2stepinit(t:Token, salt_size:Int, difficulty:Int, output:Int) -> Token {
+        // 1. Generate random bytes equal to the "salt" aka phonepart size
+        var phonepart = [UInt8](repeating: 0, count: salt_size) // array to hold randoms bytes
+        
+        // Generate random bytes
+        let res = SecRandomCopyBytes(kSecRandomDefault, salt_size, &phonepart)
+        if res != errSecSuccess {
+            U.log("random byte generation failed")
+            // TODO handle error display something?
+            return t
```

Comment:
The token that is returned in an error case - it is in any way functional or what does it do?
How would the user know, that something has failed?

Reply:
In the 2step rollout, the token is created from what is displayed, then the 2nd step is started to modify the token's secret. In the failure case, there would be no dialog showing the phone's part. That is only implicit -  so you are right, it would be better to show dialog.